src/main/java/com/internship/backend/service/AuthorityService.java

```diff
@@ -1,38 +1,50 @@
 package com.internship.backend.service;
 
+import com.couchbase.client.core.error.UserNotFoundException;
+import com.internship.backend.dto.AuthorityDTO;
+import com.internship.backend.exceptions.UserDoesNotExistException;
 import com.internship.backend.model.Authority;
+import com.internship.backend.model.Users;
 import com.internship.backend.repository.AuthorityRepository;
-import jakarta.persistence.EntityNotFoundException;
+
+import com.internship.backend.repository.UserRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class AuthorityService {
 
     @Autowired
-    private AuthorityRepository authorityRepository;
+    private UserRepository userRepository;
 
+    @Autowired
+    private AuthorityRepository authorityRepository;
 
-    public Authority add(Authority authority){
-        authorityRepository.save(authority);
-        return authority;
-    }
-
-    public void delete(int authorityId){
+    @Autowired
+    private IdGeneratorService idGeneratorService;
 
-        if(!authorityRepository.existsById(authorityId))
-            throw new EntityNotFoundException("User not found");
+    Logger logger = LoggerFactory.getLogger(AuthorityService.class);
 
-        authorityRepository.deleteById(authorityId);
+    public Authority createAuthority(Integer userId, Authority authority) throws UserDoesNotExistException {
+        Users user = userRepository.findById(userId).orElseThrow(()->new UserDoesNotExistException("User not found"));
```

Comment:
Could we test what will happen if the user is not found in  DB with an unit test? Will this method add a new user if not found?